lib/brakeman/app_tree.rb

```diff
@@ -10,10 +12,47 @@ def self.from_options(options)
       # Convert files into Regexp for matching
       init_options = {}
       if options[:skip_files]
-        init_options[:skip_files] = Regexp.new("(?:" << options[:skip_files].map { |f| Regexp.escape f }.join("|") << ")$")
+        skip_files_escaped = options[:skip_files].map do |f|
+          # If path ends in a file separator then we assume it is a path rather
+          # than a filename.
+          if f.end_with?(File::SEPARATOR)
+            # If path starts with a file separator then we assume that they
+            # want the project relative path to start with this path.
+            if f.start_with?(File::SEPARATOR)
+              "\\A#{Regexp.escape f}"
+            # If it ends in a file separator, but does not begin with a file
+            # separator then we assume the path can match any part of the project
+            # relative path.
+            else
+              Regexp.escape f
+            end
+          else
+            "#{Regexp.escape f}\\z"
+          end
+        end
+        init_options[:skip_files] = Regexp.new("(?:" << skip_files_escaped.join("|") << ")")
       end
+
       if options[:only_files]
-        init_options[:only_files] = Regexp.new("(?:" << options[:only_files].map { |f| Regexp.escape f }.join("|") << ")")
+        only_files_escaped = options[:only_files].map do |f|
+          # If it ends in a file separator then we assume it is a path rather
```

Comment:
This logic is duplicated once for each option. It could probably be folded down into a single method `self.regexes_for_paths` or some such thing.